src/Dev/Deprecation.php

```diff
@@ -163,8 +183,59 @@ protected static function get_called_method_from_trace($backtrace, $level = 1)
         if ($level == 4 && ($backtrace[$newLevel]['class'] ?? '') === InjectionCreator::class) {
             $newLevel = $newLevel + 4;
         }
+        // handle noticeWithNoReplacment()
+        foreach ($backtrace as $trace) {
+            if (($trace['class'] ?? '') === Deprecation::class
+                && ($trace['function'] ?? '') === 'noticeWithNoReplacment'
+            ) {
+                $newLevel = $newLevel + 1;
+                break;
+            }
+        }
         $called = $backtrace[$newLevel] ?? [];
-        return ($called['class'] ?? '') . ($called['type'] ?? '') . ($called['function'] ?? '');
+        return $called;
+    }
+
+    private static function isCalledFromSupportedCode(array $backtrace): bool
+    {
+        $called = Deprecation::get_called_from_trace($backtrace, 1);
+        $file = $called['file'] ?? '';
+        if (!$file) {
+            return false;
+        }
+        return Deprecation::fileIsInSupportedModule($file);
+    }
+
+    /**
+     * Check whether a file (path to file) is in a supported module
+     */
+    public static function fileIsInSupportedModule(string $file): bool
+    {
+        // Cache the supported modules list
+        if (count(Deprecation::$supportedModuleDirectories) === 0) {
+            // Manually load the supported modules list rather than use MetaData::getAllRepositoryMetaData()
+            // because we do not want to make a network request which could slow down a website
+            // While there is a small risk of the list being out of date, there is minimal downside to this
+            $path = Path::join(BASE_PATH, 'vendor/silverstripe/supported-modules/repositories.json');
+            if (!file_exists($path)) {
+                throw new RuntimeException('Could not find supported modules list');
```

Comment:
Just log it, don't throw an exception. This isn't a critical path and shouldn't result in errors if (for example) we can't read that file while reading config at the beginning of a request. Same with the error below.

Reply:
We don't use logging elsewhere for errors?

There's also no definition anywhere of what is and what isn't a "critical path", and I'd also rather not have one as that introduces a grey area. Broken is broken IMO

We have now have a tagged version of supported-modules as a dependency. If renamed repositories.json in there then we've broken something. Most proper way of doing this would be to introduce new API into supported modules to read repositories.json without making curl request to get the latest version which I can do if you want

Reply:
Ha just saw your commend below was exactly this

Reply:
> There's also no definition anywhere of what is and what isn't a "critical path", and I'd also rather not have one as that introduces a grey area

Apologies, I wasn't trying to imply that we have some policy and definition of "critical path". I was using that terminology to avoid saying in long form "This isn't code that is required for successful request-response loops - nothing here should affect end-user experiences of the website in any way, so we shouldn't be throwing exceptions here"

> We don't use logging elsewhere for errors?

Do you know that, or are you guessing that? I think we do. But if we don't, then I recommend we demote this to a warning and log it.

Reply:
Closer to know rather than guess. I searched around for Logger/LoggerInterface in our code base and it looks like it's primarily used for "stuff happened" rather than "stuff went wrong"